<launch>

    TYPHHOON + OPTIC FLOW
    <arg name="publish-images" default="false"/>
    <!--<arg name="world" default="Seville.world"/>-->
    <!--<arg name="world" default="flat.world"/>-->
    <arg name="world" default="forest_plane_fine.world"/>
    <arg name="vehicle" default="typhoon_copy"/>
    <arg name="use_hgt_compensation" default="true"/>
    <arg name="use_optic_flow" default="true"/>
    <arg name="comp_yaw" default="false"/>
    <arg name="of_im_topic_name" default="/resize_img/image"/>
    <arg name="of_im_info_name" default="/resize_img/camera_info"/>
    <arg name="gui" default="True"/>
    <arg name="save_images" default="False"/>
    <arg name="main_image_width" default="235"/>
    <arg name="main_image_height" default="150"/>

    <include file="$(find lfdrone)/src/torf/launch/gimbal_drone.launch">
        <arg name="world" value="$(arg world)"/>
        <arg name="vehicle" value="$(arg vehicle)"/>
        <!--<arg name="verbose" value="true"/>-->
        <!--<arg name="interactive" value="false"/>-->
        <!--<arg name="respawn_gazebo" value="true"/>-->
        <!--<arg name="respawn_mavros" value="false"/>-->
        <arg name="publish-images" value="$(arg publish-images)"/>
        <arg name="use_hgt_compensation" value="$(arg use_hgt_compensation)"/>
        <arg name="comp_yaw" value="$(arg comp_yaw)"/>
        <arg name="of_im_topic_name" value="$(arg of_im_topic_name)"/>
        <arg name="of_im_info_name" value="$(arg of_im_info_name)"/>
        <arg name="gui" value="$(arg gui)"/>
        <arg name="main_image_width" default="$(arg main_image_width)"/>
        <arg name="main_image_height" default="$(arg main_image_height)"/>
    </include>

    SWEEP MISSION
    <arg name="duration_out" default="15.0"/>
    <arg name="mission_hgt" default="5.0"/>
    <arg name="outbound_hgt" default="5.0"/>
    <arg name="hdg_out" default="0.0"/>
    <arg name="angle_of_attack" default="70.0"/>
    <arg name="search_angle_of_attack" default="70.0"/>
    <arg name="vel_tgt" default="1.0"/>
    <arg name="multiplier" default="5.0"/>
    <arg name="amplitude" default="8.0"/>
    <arg name="multi_process" default="true"/>
    <arg name="outbound_images" default="150"/>
    <arg name="cwssim_im_topic" default="/resize_img/image"/>
    <arg name="x_offset" default="0.0"/>
    <arg name="slowdown_factor" default="0.7"/>
    <arg name="t_turnaround" default="5"/>
    <arg name="t_familiar" default="1.0"/>   TODO make sure this is in real counterpart
    <arg name="outbound_configuration" default="fixed_heading"/>   TODO make sure this is in real counterpart
    <arg name="mission_type" default="active"/>   TODO make sure this is in real counterpart
    <arg name="outbound_sin_freq" default="0.1"/>   TODO make sure this is in real counterpart
    <arg name="cwssim_thresh" default="0.82"/>   TODO make sure this is in real counterpart
    <arg name="flip_images" default="true"/>
    <arg name="hdg_in_offset" default="0.0"/>
    <arg name="winsize" default="7"/>
    <arg name="levels" default="5"/>
    <arg name="resize_h" default="None"/>
    <arg name="resize_w" default="None"/>
    <arg name="close_to_end_thresh_high" default="25"/>
    <arg name="close_to_end_thresh_low" default="8"/>

    <node pkg="lfdrone" type="torf_ros.py" name="cwssim_node" output="screen" required="true"
            args="
                --mission_type $(arg mission_type)
                --duration_out $(arg duration_out)
                --mission_hgt  $(arg mission_hgt)
                --outbound_hgt  $(arg outbound_hgt)
                --hdg_out $(arg hdg_out)
                --x_offset $(arg x_offset)
                --angle_of_attack $(arg angle_of_attack)
                --search_angle_of_attack $(arg search_angle_of_attack)
                --vel_tgt $(arg vel_tgt)
                --multiplier $(arg multiplier)
                --amplitude $(arg amplitude)
                --multi_process $(arg multi_process)
                --outbound_images $(arg outbound_images)
                --camera_topic $(arg cwssim_im_topic)
                --angle_of_attack $(arg angle_of_attack)
                --slowdown_factor $(arg slowdown_factor)
                --t_turnaround $(arg t_turnaround)
                --t_familiar $(arg t_familiar)
                --outbound_configuration $(arg outbound_configuration)
                --outbound_sin_freq $(arg outbound_sin_freq)
                --cwssim_thresh $(arg cwssim_thresh)
                --t_familiar $(arg t_familiar)
                --outbound_configuration $(arg outbound_configuration)
                --outbound_sin_freq $(arg outbound_sin_freq)
                --cwssim_thresh $(arg cwssim_thresh)
                --flip_images $(arg flip_images)
                --hdg_in_offset $(arg hdg_in_offset)
                --winsize $(arg winsize)
                --im_w $(arg main_image_width)
                --im_h $(arg main_image_height)
                --levels $(arg levels)
                --close_to_end_thresh_high $(arg close_to_end_thresh_high)
                --close_to_end_thresh_low $(arg close_to_end_thresh_low)
              "
        />

  IMAGE WATCHDOG NODE
    <arg name="im_watchdog" default="true"/>
    <node pkg="lfdrone" type="image_watchdog_node.py" name="image_watchdog" output="screen" required="false" if="$(arg im_watchdog)"/>

    ROSBAG
    <arg name="bag_prefix" default="sweep_mission"/>
    <include file="$(find lfdrone)/src/torf/launch/bag_cwssim_mission.launch">
        <arg name="bag_prefix" value="$(arg bag_prefix)"/>
        <arg name="save_images" value="$(arg save_images)"/>
    </include>


</launch>